### PR TITLE
Fallback to IRadio in binder_voicecall_answer and merge calls lists

### PR DESCRIPTION
**Pull request by NotKit**
MTK IMS implementation expects the radio client to answer the call via the regular IRadio interface, even though the call never appears in IRadio's getCurrentCalls list

This seems to require extra handling in ofono-binder-plugin not to drop ext-initiated calls when they are not in getCurrentCalls response, so I'm trying to implement the call list merging before setting the list based on either IRadio or ext responses.

### CODE REVIEW

File: src/binder_voicecall.c [hunk 1]
```diff
@@ -300,6 +300,54 @@ binder_voicecall_info_ext_new(
...
+    for (GSList* l = self->calls; l; l = l->next) {
+        const BinderVoiceCallInfo* call = l->data;
+
+        if (!!call->ext == add_ext) {
+            if (binder_voicecall_list_find_call_with_id(new_list, call->oc.id))
+                continue;
+
+            new_list = g_slist_insert_sorted(new_list,
+                g_slice_dup(BinderVoiceCallInfo, call),
+                binder_voicecall_info_compare);
```
Comment (monich) [inline src/binder_voicecall.c:344]: Is there a reason not to break the loop when the call is found? Just for the sake of sanity, I understand that in practice there will rarely be more than once call in the list.

Comment (NotKit) [reply src/binder_voicecall.c:344]: The way I thought it was to skip the particular call that exists in both lists, but still check if there are others that are not. Do you suggest to exit early if a single call is found in both lists?

Comment (monich) [reply src/binder_voicecall.c:344]: Ah, you're right.

**Review by monich — APPROVED** [review]
LGTM

**Comment by monich** [comment]
I'm going to merge this one if there are no objections...

**Comment by mlehtima** [comment]
Should be fine to merge.

**Pull request merged**